src/Utils/ProtectedRoutes.tsx

```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+
+interface ProtectedRoutesProps {
+  component: React.FunctionComponent;
```

Comment:
Can you use [`children`](https://reactjs.org/docs/composition-vs-inheritance.html#containment) instead?

Reply:
Yes, rendering children also saves an extra step of unmounting the current component and mounting a new component. 
Thanks